nativelib/src/main/scala/scala/scalanative/native/Word.scala

```diff
@@ -0,0 +1,43 @@
+package scala.scalanative
+package native
+
+import scala.runtime.BoxesRunTime._
+import scala.reflect.ClassTag
+import scalanative.runtime._
+import scalanative.runtime.Intrinsics._
+import scalanative.runtime.Boxes._
+
+final class Word(private[scalanative] val rawWord: RawWord) {
+  @alwaysinline def toInt: Int     = castRawWordToInt(rawWord)
+  @alwaysinline def toUInt: UInt   = castRawWordToInt(rawWord).toUInt
+  @alwaysinline def toUWord: UWord = new UWord(rawWord)
+
+  @alwaysinline def +(other: Word): Word =
+    new Word(addRawWords(rawWord, other.rawWord))
+  @alwaysinline def -(other: Word): Word =
+    new Word(subRawWords(rawWord, other.rawWord))
+  @alwaysinline def *(other: Word): Word =
+    new Word(multRawWords(rawWord, other.rawWord))
+  @alwaysinline def /(other: Word): Word =
+    new Word(divRawWords(rawWord, other.rawWord))
+
+  override def hashCode: Int =
+    java.lang.Long.hashCode(Word.wordToLong(this))
+
+  override def equals(other: Any): Boolean =
+    (this eq other.asInstanceOf[AnyRef]) || (other match {
+      case other: Word =>
+        other.rawWord == rawWord
+      case _ =>
+        false
+    })
+
+  override def toString(): String = castRawWordToLong(rawWord).toString
+}
+
+object Word {
+  @alwaysinline implicit def intToWord(int: Int): Word =
+    new Word(castIntToRawWord(int))
+  @alwaysinline implicit def wordToLong(word: Word): Long =
+    castRawWordToLong(word.rawWord)
```

Comment:
Needs conversions from the rest of integer numeric types as well.

Reply:
The idea here was to only have "safe" conversions that will work correctly on all platforms, so that's why I don't have a conversion from `Long` to `Word` and instead `.toWord` has to be explicitly called. I'll add in conversions for smaller integer types to words though.

Reply:
Lets for with explicit conversions only then. This 100% safe and explicit. Unsigned integers for example don't have implicit conversions.

Reply:
I tried making this change to have everything be explicit, but there are a lot of places where we perform operations on words with literal numbers (for example `stackalloc` with a count), so making everything explicit would end up forcing users to put a lot of `toWord`s everywhere. I decided to drop the conversion away from `Word` since that's less common, and have added conversions to `Word` from all integer types with size <= 32 bits. This way, the behavior matches how Scala handles conversions for regular numeric types.